src/main/java/io/airlift/slice/Slice.java

```diff
@@ -540,9 +540,32 @@ public float getFloat(int index)
     public double getDouble(int index)
     {
         checkIndexLength(index, SIZE_OF_DOUBLE);
+        return getDoubleUnchecked(index);
+    }
+
+    double getDoubleUnchecked(int index)
+    {
         return unsafe.getDouble(base, address + index);
     }
 
+    /**
+     * Gets an array of {@code length} 64-bit doubles from the absolute
+     * {@code index} in this buffer.
+     *
+     * @throws IndexOutOfBoundsException if the specified {@code index} is less than {@code 0} or
+     * {@code index + 8 * length} is greater than {@code this.length()}
+     */
+    public double[] getDoubles(int index, int length)
```

Comment:
It'd be more flexible to let the caller pass the output array + offset. This would avoid extra allocation and copying if the results need to be stitched with something else.

Reply:
I’d agree and say that the convention should be applied to the methods in SliceInput classes for the same reason: internal allocations make it impossible for callers to avoid them.

Similarly, the SliceOutput API should allow passing an offset + length along with the buffer so that sub range copies aren’t necessary to use the batch API.